pkg/apis/core/validation/shoot.go

```diff
@@ -1101,31 +1110,46 @@ func ValidateAPIServerRequests(requests *core.APIServerRequests, fldPath *field.
 	return allErrs
 }
 
-func validateEncryptionConfig(encryptionConfig *core.EncryptionConfig, defaultEncryptedResources sets.Set[string], fldPath *field.Path) field.ErrorList {
-	allErrs := field.ErrorList{}
+func validateEncryptionConfig(encryptionConfig *core.EncryptionConfig, defaultEncryptedResources []schema.GroupResource, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs       = field.ErrorList{}
+		seenResources []schema.GroupResource
```

Comment:
We can still use the set here?
```suggestion
		seenResources = sets.New[schema.GroupResource]()
```